.github/workflows/gh-build-docs.yml

```diff
@@ -0,0 +1,148 @@
+  build:
+    name: Build docs
+    # The build stage could fail but we want the CI to keep moving.
+    if: ${{ github.repository_owner == 'nvidia' && always() }}
+    # WAR: Building the doc currently requires a GPU (NVIDIA/cuda-python#326)
+    runs-on: linux-amd64-gpu-t4-latest-1-testing
+    #runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout ${{ github.event.repository.name }}
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up miniforge
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          activate-environment: cuda-python-docs
+          environment-file: ./cuda_python/docs/environment-docs.yml
+          miniforge-version: latest
+          conda-remove-defaults: "true"
+          python-version: 3.12
+
+      # WAR: Building the doc currently requires CTK installed (NVIDIA/cuda-python#326)
+      - name: Set up mini CTK
+        uses: ./.github/actions/fetch_ctk
+        continue-on-error: false
+        with:
+          host-platform: linux-64
+          cuda-version: ${{ inputs.build_ctk_ver }}
+
+      - name: Set environment variables
+        run: |
+          PYTHON_VERSION_FORMATTED="312"  # see above
+          REPO_DIR=$(pwd)
+
+          # make outputs from the previous job as env vars
+          echo "CUDA_CORE_ARTIFACT_NAME=cuda-core-python${PYTHON_VERSION_FORMATTED}-linux-64-${{ github.sha }}" >> $GITHUB_ENV
+          echo "CUDA_CORE_ARTIFACTS_DIR=$(realpath "$REPO_DIR/cuda_core/dist")" >> $GITHUB_ENV
+          echo "CUDA_BINDINGS_ARTIFACT_NAME=cuda-bindings-python${PYTHON_VERSION_FORMATTED}-cuda${{ inputs.build_ctk_ver }}-linux-64-${{ github.sha }}" >> $GITHUB_ENV
+          echo "CUDA_BINDINGS_ARTIFACTS_DIR=$(realpath "$REPO_DIR/cuda_bindings/dist")" >> $GITHUB_ENV
+
+      - name: Download cuda.bindings build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ env.CUDA_BINDINGS_ARTIFACT_NAME }}
+          path: ${{ env.CUDA_BINDINGS_ARTIFACTS_DIR }}
+
+      - name: Display structure of downloaded cuda.bindings artifacts
+        run: |
+          pwd
+          ls -lahR $CUDA_BINDINGS_ARTIFACTS_DIR
+
+      - name: Download cuda.core build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ env.CUDA_CORE_ARTIFACT_NAME }}
+          path: ${{ env.CUDA_CORE_ARTIFACTS_DIR }}
+
+      - name: Display structure of downloaded cuda.core build artifacts
+        run: |
+          pwd
+          ls -lahR $CUDA_CORE_ARTIFACTS_DIR
+
+      - name: Install all packages
+        run: |
+          pushd "${CUDA_BINDINGS_ARTIFACTS_DIR}"
+          pip install *.whl
+          popd
+
+          pushd "${CUDA_CORE_ARTIFACTS_DIR}"
+          pip install *.whl
+          popd
+
+      - name: Build all (latest) docs
+        id: build
+        run: |
+          pushd cuda_python/docs/
+          ./build_all_docs.sh latest-only
```

Comment:
Note: This workflow currently only builds the "latest" docs; those with release versions (ex: cuda.bindings 12.6.2) need a separate step that we should add in another PR (once we figure out the release workflow).

Reply:
I think the logic should be as simple as (I could be wrong!)
- Is the doc CI triggered by PRs or merging to main?
   - Yes: Only update `latest`
   - No: Assert this is triggered by a release. Update both `latest` and the latest release version.
      - TODO: check if we have a way to declare if we only want to update a certain component's docs.